### DISCUSSION TITLE
Unable to fetch metrics

### DISCUSSION THREAD

**Discussion by MohithSarvepalli** [kedacore/keda#4043] [Q&A / Need Help] [answered]
I have been trying to get metrics from prometheus, after writing my scaled yml, it says 
`Warning  FailedGetExternalMetric  9s (x12 over 2m54s)  horizontal-pod-autoscaler  unable to get external metric default/s0-prometheus-cpu_usage/&LabelSelector{MatchLabels:map[string]string{scaledobject.keda.sh/name: prometheus-scaledobject,},MatchExpressions:[]LabelSelectorRequirement{},}: unable to fetch metrics from external metrics API: rpc error: code = Unknown desc
= error when getting metric values no matching metrics found for s0-prometheus-cpu_usage`

**Comment by MohithSarvepalli** [comment, marked as answer]
Resolved, Thank you

**Reply by tomkerkhove** [reply]
Would you mind elaborating how you fixed it for other people who might have the same issue?

**Reply by MohithSarvepalli** [reply]
I fixed it by removing /graph from the prometheus ip that I gave earlier and fixed the query that I used, I forgot to add sum() and when ever new pod used to spin up, metrics collection was stopped by hpa because I got two pods and two values.

**Reply by sriteja124** [reply]
could you please provide your scaledobject definition, i'm also facing similar kind of issue
